src/global/helpers/arrow-navigation.ts

```diff
@@ -0,0 +1,46 @@
+import getDocumentWritingMode from './get-document-writing-mode';
+
+/**
+ * Check if the key pressed is among those allowed for navigation.
+ * @param event The keyboard event to check.
+ */
+export function isArrowKeyPressed(event: KeyboardEvent): boolean {
+  return ['ArrowRight', 'ArrowLeft', 'ArrowUp', 'ArrowDown'].includes(event.key);
+}
+
+/**
+ * Calculate the index of the next element based on the given offset
+ * @param currentIndex The index of the current element.
+ * @param size The size of the array.
+ * @param offset The amount to move by.
+ */
+function goToElement(currentIndex: number, size: number, offset: number): number {
```

Comment:
maybe something like that to be more specific?
```suggestion
function calcIndexInRange(currentIndex: number, maxIndex: number, offset: number): number {
```